prov/verbs/src/verbs_info.c

```diff
@@ -428,17 +429,18 @@ static inline int fi_ibv_get_qp_cap(struct ibv_context *ctx,
 		goto err1;
 	}
 
+	qp_type = (info->ep_attr->type != FI_EP_DGRAM) ?
+			    IBV_QPT_RC : IBV_QPT_UD;
+
 	memset(&init_attr, 0, sizeof init_attr);
 	init_attr.send_cq = cq;
 	init_attr.recv_cq = cq;
 	init_attr.cap.max_send_wr = fi_ibv_gl_data.def_tx_size;
 	init_attr.cap.max_recv_wr = fi_ibv_gl_data.def_rx_size;
 	init_attr.cap.max_send_sge = fi_ibv_gl_data.def_tx_iov_limit;
 	init_attr.cap.max_recv_sge = fi_ibv_gl_data.def_rx_iov_limit;
-	init_attr.cap.max_inline_data = fi_ibv_find_max_inline(pd, ctx);
-
-	init_attr.qp_type = (info->ep_attr->type != FI_EP_DGRAM) ?
-			    IBV_QPT_RC : IBV_QPT_UD;
+	init_attr.cap.max_inline_data = fi_ibv_find_max_inline(pd, ctx, qp_type);
```

Comment:
This needs to be cached.  The overhead of allocating a bunch of QPs prior to allocating the real one is significant.  This is separate from this patch, however.

Reply:
Looking at the upstream code, it looks like this is being cached.  Ignore me.  :)